### ISSUE TITLE
Algorithm.function to update the leaf nodes in the MerkleTree

### ISSUE THREAD

**Issue by AlvinKuruvilla** [antouhou/rs-merkle#26]
First off I want to thank you for making this crate. I had one potential suggestion to improve the crate. I feel that it would be useful to have a helper function that enables a leaf node in a given tree to be updated. I tried looking in the documentation to see if such functionality already existed but I couldn't find anything. 
My current workaround for this issue is to update the set of leaves and create an entirely new MerkleTree from there, however, this is very wasteful especially when dealing with large leaf counts. 
Do you have any suggestions for a better way to do something like this, or if a function could be designed to do this?

**Comment by AlvinKuruvilla** [comment]
Of course, thanks for getting back to me. If you could give me some direction on how you would go about implementing a feature like this, I could try writing a PR for it

**Comment by Pompey21** [comment]
+1: having this feature would be absolutely amazing!!
